src/providers/oauth2/types.ts

```diff
@@ -1,35 +1,36 @@
-import { OAuth2ProviderTypeValues } from "../../schema/providers/oauth2/types.ts";
+import { ManifestOAuth2ProviderSchema } from "../../manifest/manifest_schema.ts";
 
-export type OAuth2ProviderIdentitySchema = {
-  "url": string;
-  "account_identifier": string;
-  "headers"?: {
-    [key: string]: string;
-  };
+export type OAuth2ProviderDefinitionArgs = ManifestOAuth2ProviderSchema & {
+  /** A unique name for the provider */
+  provider_key: string;
 };
 
+/**
+ * TODO: The type system here could be improved one more then one provider type (CUSTOM) is available
+ * provider_name, authorization_url, token_url, identity_config and authorization_url_extras
```

Comment:
In theory these should be required but I fear that making these fields `required` might break some apps

Reply:
How did you determine that these fields are required? Checked the backend schema or some other way?
Can you confirm that not providing these fields works? I.e. can you deploy an app and pass manifest validation if you remove these fields?

Reply:
The comments for each of those field mentioned that they we required for `CUSTOM` provider types, I am assuming they are true

But since those fields are currently `not required` I don't think we can make them `required` without potentially breaking apps

Reply:
"Breaking" from a static-type-check perspective, that is true, however, if the backend requires these fields anyways, then that is much less a concern, I think, because no _real_ app would be able to define a provider without them.

That's why I think we should test this out on a real app 😄